src/test/java/org/opensearch/security/dlic/rest/api/ActionGroupsApiTest.java

```diff
@@ -340,6 +364,93 @@ public void testActionGroupsApi() throws Exception {
         Assert.assertTrue(permissions.contains("READ_UT"));
     }
 
+    @Test
+    public void testActionGroupsApiForRestAdmin() throws Exception {
+        setupWithRestRoles();
+        rh.sendAdminCertificate = false;
+        // create index
+        setupStarfleetIndex();
+        final Header restApiAdminHeader = encodeBasicHeader("rest_api_admin_user", "rest_api_admin_user");
+
+        // add user picard, role starfleet, maps to opendistro_security_role_starfleet
+        addUserWithPassword("picard", "picard", new String[] { "starfleet" }, HttpStatus.SC_CREATED);
+        checkReadAccess(HttpStatus.SC_OK, "picard", "picard", "sf", "_doc", 0);
+        // TODO: only one doctype allowed for ES6
+        // checkReadAccess(HttpStatus.SC_OK, "picard", "picard", "sf", "public", 0);
+        checkWriteAccess(HttpStatus.SC_FORBIDDEN, "picard", "picard", "sf", "_doc", 0);
+        // TODO: only one doctype allowed for ES6
+        //checkWriteAccess(HttpStatus.SC_OK, "picard", "picard", "sf", "public", 0);
+        verifyGetForSuperAdmin(new Header[] {restApiAdminHeader});
+        verifyDeleteForSuperAdmin(new Header[]{restApiAdminHeader}, false);
+        verifyPutForSuperAdmin(new Header[]{restApiAdminHeader}, false);
+        verifyPatchForSuperAdmin(new Header[]{restApiAdminHeader}, false);
+    }
+
+    @Test
+    public void testActionGroupsApiForActionGroupsRestApiAdmin() throws Exception {
+        setupWithRestRoles();
+        rh.sendAdminCertificate = false;
+        // create index
+        setupStarfleetIndex();
+        final Header restApiAdminActionGroupsHeader = encodeBasicHeader("rest_api_admin_actiongroups", "rest_api_admin_actiongroups");
+
+        // add user picard, role starfleet, maps to opendistro_security_role_starfleet
+        addUserWithPassword("picard", "picard", new String[] { "starfleet" }, HttpStatus.SC_CREATED);
+        checkReadAccess(HttpStatus.SC_OK, "picard", "picard", "sf", "_doc", 0);
+        // TODO: only one doctype allowed for ES6
```

Comment:
Looks like there are a couple of TODOs, were you planning on addressing these?

Reply:
Doctypes were removed in OpenSearch 2.0 https://github.com/opensearch-project/opensearch/issues/1940

So I wouldn't worry about legacy different doctype scenarios.

Reply:
Well I just moved code for now :-) Sure can remove todo